model/task/task.go

```diff
@@ -1071,6 +1088,8 @@ func (t *Task) MarkAsContainerDeallocated(ctx context.Context, env evergreen.Env
 	t.ContainerAllocated = false
 	t.DispatchTime = utility.ZeroTime
 	t.LastHeartbeat = utility.ZeroTime
+	t.ContainerAllocatedTime = time.Time{}
```

Comment:
`utility.ZeroTime`?

Reply:
This one is kind of weird because this is an optional time field (probaby all of them should be optional, but nothing I can do about them now). Go will only omitempty if it's the Go zero time, not Unix zero time.

I guess we could just use the Unix zero time for everything since the precedence was set, but I don't have strong feelings either way.